spark-connector/spark-common/src/main/java/org/apache/gravitino/spark/connector/iceberg/extensions/IcebergExtendedDataSourceV2Strategy.java

```diff
@@ -81,7 +84,7 @@ public Seq<SparkPlan> apply(LogicalPlan plan) {
     } else if (plan instanceof CreateOrReplaceBranch) {
       CreateOrReplaceBranch createOrReplaceBranch = (CreateOrReplaceBranch) plan;
       return IcebergCatalogAndIdentifier.buildCatalogAndIdentifier(
-              spark, createOrReplaceBranch.table())
+              spark, createOrReplaceBranch.table().toIndexedSeq())
```

Comment:
why changing `toIndexedSeq`?

Reply:
```
error: apply(LogicalPlan) in IcebergExtendedDataSourceV2Strategy cannot override apply(LogicalPlan) in GenericStrategy
  public Seq<SparkPlan> apply(LogicalPlan plan) {
                        ^
  return type scala.collection.Seq<SparkPlan> is not compatible with scala.collection.immutable.Seq<SparkPlan>

```

The `type scala.Seq` in scala2.13 represents the  `scala.collection.immutable.Seq`, `toIndexedSeq` can change the return type to `scala.collection.immutable.Seq`.
and `IndexedSeq` is a sub `trait` of `scala.collection.immutable.Seq`